src/main/java/net/openhft/chronicle/queue/impl/single/TableDirectoryListing.java

```diff
@@ -41,21 +41,32 @@ final class TableDirectoryListing extends AbstractCloseable implements Directory
     private final TableStore<?> tableStore;
     private final Path queuePath;
     private final ToIntFunction<String> fileNameToCycleFunction;
+    private final boolean readOnly;
     private volatile LongValue maxCycleValue;
     private volatile LongValue minCycleValue;
     private volatile LongValue modCount;
     private long lastRefreshTimeMS = 0;
 
+    @Deprecated(/* to be removed in x.25 */)
```

Comment:
Propose we create a separate `ReadOnlyTableDirectoryListing` instead of adding all the extra branching in. They share almost no logic, and the `readOnly` flag only adds undue complexity to the writable version.

Reply:
Done, although with inheritance ;)